### PR TITLE
implement callback behavior for AsyncStorage.getAllKeys()

### LINKED ISSUE #399: AsyncStorage.getAllKeys() does not behave like React Native's API
The documentation for React Native's AsyncStorage API's `getAllKeys()` says that the function should take a callback as the first argument which is a "Function that will be called the keys found and any error."

However, the react-native-web version does not accept any parameters.

While it is possible to use the API with the promise object returned by the function, it probably should also handle the callback, as documented.

I'll open a PR for this in the morning with a fix, unless there is some reason for the omission.

### PR DESCRIPTION
**Pull request by aaroncraig10e**
Resolves #399

### CODE REVIEW

**Comment by necolas** [comment]
All the functions support the callback style usage in RN: https://facebook.github.io/react-native/docs/asyncstorage.html

File: src/apis/AsyncStorage/__tests__/index-test.js [hunk 1]
```diff
@@ -21,25 +21,92 @@ const waterfall = (fns, cb) => {
...
+    beforeEach(async () => {
+      window.localStorage = mockLocalStorage;
+      await AsyncStorage.setItem('UID123', JSON.stringify(UID123_object));
+    });
+
+    afterEach(() => {
+      window.localStorage = originalLocalStorage;
+    });
+
+    test('should have same Promise behavior as react-native', async () => {
```
Comment (necolas) [inline src/apis/AsyncStorage/__tests__/index-test.js]: Please can you use more descriptive test names than "same as react-native"

File: src/apis/AsyncStorage/index.js [hunk 2]
```diff
@@ -40,8 +40,17 @@ class AsyncStorage {
           const key = window.localStorage.key(i);
           keys.push(key);
         }
+
+        if (cb) {
+          cb(null, keys);
+        }
+
         resolve(keys);
```
Comment (necolas) [inline src/apis/AsyncStorage/index.js]: Is the expected behaviour that the callback is called AND the promise resolved?

Comment (aaroncraig10e) [inline src/apis/AsyncStorage/index.js]: That's my understanding from the documentation (and using AsyncStorage in native).

**Comment by nthtran** [comment]
Would be great to support callbacks. Libs like `redux-persist` rely on this behaviour.

**Comment by aaroncraig10e** [comment]
@nthtran That's how I discovered this issue, actually -- we're using `redux-persist` and in fact, it breaks without this patch.

**Comment by aaroncraig10e** [comment]
@necolas any chance of getting this merged in?

**Comment by necolas** [comment]
I'll accept a PR that adds the callback interface to every function that supports it in the RN AsyncStorage API

**Comment by aaroncraig10e** [comment]
Fair enough -- I'll get that together over the next week or so.

**Pull request closed without merging**